File: FLUTTER/lib/features/smart_deal/data/models/promotion/PromotionPointDTO.dart
class PromotionPointDTO {
  String id;
  String title;
  String description;
  List<String> discountType;
  double discountValue;
  DateTime? startDate;
  DateTime? endDate;
  bool isActive;
  List<String> applicableService;
  double minValue;
  List<String> customerType;
  String createdBy;
  List<String> packageName;
  String code;
  int points;

  PromotionPointDTO({
    required this.id,
    required this.title,
    required this.description,
    required this.discountType,
    required this.discountValue,
    this.startDate,
    this.endDate,
    required this.isActive,
    required this.applicableService,
    required this.minValue,
    required this.customerType,
    required this.createdBy,
    required this.packageName,
    required this.code,
    required this.points,
  });

  factory PromotionPointDTO.fromJson(Map<String, dynamic> json) {
    return PromotionPointDTO(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      discountType: List<String>.from(json['discountType'] ?? []),
      discountValue: (json['discountValue'] as num).toDouble(),
      startDate: json['startDate'] != null ? DateTime.tryParse(json['startDate']) : null,
      endDate: json['endDate'] != null ? DateTime.tryParse(json['endDate']) : null,
      isActive: json['isActive'] ?? false,
      applicableService: List<String>.from(json['applicableService'] ?? []),
      minValue: (json['minValue'] as num).toDouble(),
      customerType: List<String>.from(json['customerType'] ?? []),
      createdBy: json['createdBy'],
      packageName: List<String>.from(json['packageName'] ?? []),
      code: json['code'],
      points: json['points'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'description': description,
      'discountType': discountType,
      'discountValue': discountValue,
      'startDate': startDate?.toIso8601String(), // Trả về null nếu startDate null
      'endDate': endDate?.toIso8601String(), // Trả về null nếu endDate null
      'isActive': isActive,
      'applicableService': applicableService,
      'minValue': minValue,
      'customerType': customerType,
      'createdBy': createdBy,
      'packageName': packageName,
      'code': code,
      'points': points,
    };
  }
}

File: BACKEND/smartdeals-service/src/main/java/data/smartdeals_service/dto/promotion/PromotionPointDTO.java
package data.smartdeals_service.dto.promotion;

import com.fasterxml.jackson.annotation.JsonFormat;
import data.smartdeals_service.models.promotion.ApplicableServices;
import data.smartdeals_service.models.promotion.CustomerType;
import data.smartdeals_service.models.promotion.DiscountType;
import data.smartdeals_service.models.promotion.PackageName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class PromotionPointDTO {
    private String id;
    private String title;
    private String description;
    private List<DiscountType> discountType;
    private BigDecimal discountValue;
    private LocalDateTime startDate;
    private LocalDateTime endDate;
    private Boolean isActive;
    private List<ApplicableServices> applicableService;
    private BigDecimal minValue;
    private List<CustomerType> customerType;
    private String createdBy;
    private List<PackageName> packageName;
    private String code;
    private int points;
}
